docs/src/modules/components/Link.tsx

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable jsx-a11y/anchor-has-content */
+import * as React from 'react';
+import clsx from 'clsx';
+import { useRouter } from 'next/router';
+import NextLink, { LinkProps as NextLinkProps } from 'next/link';
+import MuiLink, { LinkProps as MuiLinkProps } from '@material-ui/core/Link';
+import { useUserLanguage } from 'docs/src/modules/utils/i18n';
+
+type NextLinkComposedProps = Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, 'href'> &
+  Omit<NextLinkProps, 'href' | 'as'> & {
+    to: NextLinkProps['href'];
+    linkAs?: NextLinkProps['as'];
+    href?: any;
+  };
+
+export const NextLinkComposed = React.forwardRef<HTMLAnchorElement, NextLinkComposedProps>(
+  (props, ref) => {
```

Comment:
```suggestion
  (props, ref) => {
```
Could you give this a name for readable stack traces and devtools readability? Might make sense  to add a lint rule if this mistake is repeated frequently.